### PR TITLE
Fix date time validations

### PR DESCRIPTION
**Pull request by SandyChapman**
Looks like proper validation is not happening on date-times passed through parameters. This is happening because we first parse the values given to Javascript `Date`s and then do validation later on the parsed `Date` object. This means that anything that is able to construct a non-null `Date` object is being recognized as a valid date-time value.

This means things like "1223" and "jan 5" were being recognized as valid. This PR fixes this by adding validation prior to parsing to a Date object and adds unit test coverage for the error cases.

### CODE REVIEW

**Comment by whitlockjc** [comment]
Do you mind squashing your commits?  I'll review the rest now and update accordingly.

On a different note, do you mind checking if [sway](https://github.com/apigee-127/sway) needs the same enhancement?

File: test/2.0/test-middleware-swagger-validator.js [hunk 1]
```diff
@@ -208,50 +208,117 @@ describe('Swagger Validator Middleware v2.0', function () {
...
-        if (scenario.type === 'array') {
-          expectedMessage = 'Parameter (' + argName + ') at index 1 is not a valid integer: fake';
-        } else {
-          expectedMessage = 'Parameter (' + scenario.name + ') is not a valid ' +
-                              (_.isUndefined(scenario.format) ?
-                                 '' :
-                                 scenario.format + ' ') + scenario.type + ': ' + badValue;
-        }
+    it('should return an error for invalid parameter values based on type/format', function (done) {
+        var argName = 'arg0';
```
Comment (whitlockjc) [inline test/2.0/test-middleware-swagger-validator.js]: The extra whitespace isn't desired but if you've already squashed commits, I'm not sure I mind enough not to merge the PR.

Comment (SandyChapman) [inline test/2.0/test-middleware-swagger-validator.js]: I'll correct it. My IDE added in a copy / paste.

**Comment by SandyChapman** [comment]
I don't mind squashing. Will resubmit.

Also, it does look like Sway is affected as well: https://github.com/apigee-127/sway/blob/master/lib/helpers.js#L264

**Comment by whitlockjc** [comment]
You don't have to resubmit.  Just squash commits, force push to your repo and this PR will pick it up.

**Comment by whitlockjc** [comment]
As for Sway, do you mind submitting a PR there?  I'd really appreciate it.

**Comment by SandyChapman** [comment]
Yep, did the squash and force push (like a Jedi). Mis-communicated what I was intending to do. I'll see if I can apply to sway as well.

**Comment by whitlockjc** [comment]
One last request, update your commit message to be inline with what you did.  You enhanced/fixed date-time validation but your commit message after the squash only mentions the tests.  As for Sway, that would be great.  Sway is where we're going in the future.

**Comment by SandyChapman** [comment]
Updated the commit message. I'll try to find some time today to look into Sway.

**Comment by whitlockjc** [comment]
Thanks @SandyChapman.

**Comment by oliverhr** [comment]
This is going to be tagged for version any soon?

**Pull request merged**